cclient/response/get_neighbors.c

```diff
@@ -32,26 +37,55 @@ get_neighbors_res_t* get_neighbors_res_new() {
 
 void get_neighbors_res_free(get_neighbors_res_t* nbors) { utarray_free(nbors); }
 
-neighbor_info_t* get_neighbors_res_create_neighbor(char_buffer_t* addr, int all_trans, int invalid_trans,
-                                                   int new_trans) {
+neighbor_info_t* get_neighbors_res_create_neighbor(char_buffer_t* addr, uint32_t all_trans_num,
+                                                   uint32_t random_trans_req_num, uint32_t new_trans_num,
+                                                   uint32_t invalid_trans_num, uint32_t stale_trans_num,
+                                                   uint32_t sent_trans_num, char_buffer_t* connectionType) {
   neighbor_info_t* nb = (neighbor_info_t*)malloc(sizeof(neighbor_info_t));
   if (nb == NULL) {
     return NULL;
   }
-  nb->address = addr;
-  nb->all_trans_num = all_trans;
-  nb->invalid_trans_num = invalid_trans;
-  nb->new_trans_num = new_trans;
+
+  nb->address = (char_buffer_t*)malloc(sizeof(char_buffer_t));
```

Comment:
no need to malloc, parameters are pointers and numbers, please assign to the structure directly.